helix-core/src/main/java/org/apache/helix/HelixProperty.java

```diff
@@ -148,7 +154,12 @@ public String toString() {
     }
   }
 
-  private Stat _stat;
+  protected Stat _stat;
+
+  public HelixProperty() {
+    _record = DEFAULT_ZNRECORD;
+    _stat = DEFAULT_STAT;
+  }
```

Comment:
Again, why do we need a default constructor here?

Reply:
If we don't have this default constructor, in Message's constructor, we need to call `super(ZNRecord)`. HelixProperty defines `ZNRecord` but not `SessionAwareZNRecord`. Then if we want Message's SessionAwareZNRecord, we may override `getRecord()` to convert `ZNRecord` to `SessionAwareZNRecord`: 
```
Override
public ZNRecord getRecord() {
  return new SessionAwareZNRecord(_record);
}
```
The problem is this creates a new record from the original one `_record`. We could not access the original one: `message.getRecord().setSimpleField()`.

So we need to create SessionAwareZNRecord in Message's constructor and override parent's ZNRecord. Without the default constructor, we won't be able to do `_record = new SessionAwareZNRecord()` in Message.

Reply:
How about do this?
  public HelixProperty(SessionAwareZNRecord record, String id) {
    _record = record;
    _stat = new Stat(_record.getVersion(), _record.getCreationTime(), _record.getModifiedTime(),
        _record.getEphemeralOwner());
  }

Reply:
I know there are multiple options. But how about the one that I mentioned above?

Reply:
I thought about this one before. I think it is fine. The existing constructors are already kind of messing.. We can not still converge the other constructors to this one.